Unit tests are failing with Python 3.14

##### SUMMARY
ansible-core devel now supports Python 3.14. It looks like the unit tests fail with it: https://github.com/ansible-collections/community.proxmox/actions/runs/16589730819/job/46922234842

This could be something unrelated to this collection, but someone has to take a closer look.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
unit tests

It is possible that this is a Python 3.14 compatibility problem of included libraries (in this case, probably paramiko). I've been looking at similar surprising test failures in community.general, and there I got an exception:
```
E   Exception: Traceback (most recent call last):
E     File "/root/ansible_collections/community/general/plugins/modules/github_repo.py", line 137, in <module>
E       from github import Github, GithubException, GithubObject
E     File "/usr/local/lib/python3.14/dist-packages/github/__init__.py", line 54, in <module>
E       from . import Auth
E     File "/usr/local/lib/python3.14/dist-packages/github/Auth.py", line 41, in <module>
E       from github.InstallationAuthorization import InstallationAuthorization
E     File "/usr/local/lib/python3.14/dist-packages/github/InstallationAuthorization.py", line 47, in <module>
E       import github.NamedUser
E     File "/usr/local/lib/python3.14/dist-packages/github/NamedUser.py", line 62, in <module>
E       import github.Event
E     File "/usr/local/lib/python3.14/dist-packages/github/Event.py", line 48, in <module>
E       import github.Organization
E     File "/usr/local/lib/python3.14/dist-packages/github/Organization.py", line 93, in <module>
E       import github.CodeSecurityConfigRepository
E     File "/usr/local/lib/python3.14/dist-packages/github/CodeSecurityConfigRepository.py", line 30, in <module>
E       import github.Repository
E     File "/usr/local/lib/python3.14/dist-packages/github/Repository.py", line 189, in <module>
E       import github.Environment
E     File "/usr/local/lib/python3.14/dist-packages/github/Environment.py", line 52, in <module>
E       from github.PublicKey import PublicKey
E     File "/usr/local/lib/python3.14/dist-packages/github/PublicKey.py", line 48, in <module>
E       from nacl import encoding, public
E     File "/usr/local/lib/python3.14/dist-packages/nacl/public.py", line 16, in <module>
E       import nacl.bindings
E     File "/usr/local/lib/python3.14/dist-packages/nacl/bindings/__init__.py", line 186, in <module>
E       from nacl.bindings.crypto_secretstream import (
E       ...<15 lines>...
E       )
E     File "/usr/local/lib/python3.14/dist-packages/nacl/bindings/crypto_secretstream.py", line 14, in <module>
E       from typing import ByteString, Optional, Tuple, cast
E   ImportError: cannot import name 'ByteString' from 'typing' (/usr/lib/python3.14/typing.py)
```
`typing.ByteString` has been deprecated for some time and was removed in Python 3.14. So basically the nacl package is not compatible with Python 3.14 yet...

I guess the way to fix the CI failure is to make sure that the test is skipped if its dependency (in this case: paramiko) is not there (even though it is *supposed* to be installed in CI).

--- a/tests/unit/plugins/connection/test_proxmox_pct_remote.py
+++ b/tests/unit/plugins/connection/test_proxmox_pct_remote.py
@@ -22,6 +22,9 @@ from pathlib import Path
 from unittest.mock import patch, MagicMock, mock_open
 
 
+pytest.importorskip('paramiko')
+
+
 @pytest.fixture
 def connection():
     play_context = PlayContext()

